src/senaite/ast/upgrade/v01_00_000.py

```diff
@@ -146,15 +146,11 @@ def remove_ast_from_profiles(portal):
     """
     logger.info("Removing AST-like analyses from profiles ...")
     ast_uids = get_ast_services_uids(portal)
-    query = {
-        "portal_type": "AnalysisProfile"
-    }
-    brains = api.search(query, SETUP_CATALOG)
-    for brain in brains:
-        obj = api.get_object(brain)
-        services = obj.getRawService() or []
-        services = filter(lambda s: s not in ast_uids, services)
-        obj.setService(services)
+    profiles = portal.setup.analysisprofiles.objectValues()
+    for obj in profiles:
+        services = obj.getRawServices() or []
```

Comment:
you could use here `getServiceUIDs` instead:
https://github.com/senaite/senaite.core/blob/2.x/src/senaite/core/content/analysisprofile.py#L267-L272

Reply:
With `getRawServices()` I get the whole dict for each record, with the `hidden` setting included. So I don't loose information later when I reset the field with the filtered list.

Reply:
Ah, apologies, you are right! I've overlooked the `setService` below.